Sources/Qualification/State/AppQualifyingService.swift

```diff
@@ -97,9 +97,17 @@ final class AppQualifyingService: QualifyingService {
                     userState = .userConfirmed
                 }
             } catch {
-                sessionManager.endCurrentSession()
-                sessionManager.clearAllSessionData()
-                userState = .userFailed(error)
+                switch error {
```

Comment:
is there a reason we don't want to do this for all errors?

Reply:
A `SecureStoreError.cantDecryptData` is thrown when the local auth prompt is cancelled/dismissed. I imagine from [here](https://github.com/govuk-one-login/mobile-ios-secure-store/blob/main/Sources/SecureStore/KeyManagerService.swift#L189C19-L189C51)?

Reply:
Gotcha, and we don't want to clear the session in that instance? I think a safer approach would be to catch that specific error and go through a different path, rather than leaving an empty generic catch block.

Reply:
My understanding was that we don't want to do anything other than leaving the user on the lock screen. When the user returns to the app either from boot or background the checks are performed again.
Happy to specifically catch that error but not sure what the body would be other than `return`?

Reply:
Surely an error screen and allowing them to re-login would be better than leaving them on the Lock Screen? 🤔

Reply:
An error screen when they dismiss the local authentication prompt?

Reply:
Ah sorry, I meant.. we catch _that_ error and return (doing nothing) and handle all _other_ errors.
Rather than what we have here now, which is catch _some_ specific errors we know about and do nothing for _other_ errors including _SecureStoreError.cantDecryptData_.